compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ViewControllerBasedLifecycleOwner.uikit.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.window
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import kotlin.test.fail
+
+internal class ViewControllerBasedLifecycleOwner : LifecycleOwner {
+    private enum class Action {
+        VIEW_WILL_APPEAR,
+        VIEW_DID_DISAPPEAR,
+        APPLICATION_DID_ENTER_BACKGROUND,
+        APPLICATION_WILL_ENTER_FOREGROUND,
+        DISPOSE
+
+        // TODO: add actions for Popup and Dialog to behave like Android
+    }
+
+    private sealed interface State {
+        fun reduce(action: Action): State
+
+        class Created(
+            private val isApplicationForeground: Boolean,
+            private val lifecycle: LifecycleRegistry
+        ) : State {
+            override fun reduce(action: Action): State {
+                return when (action) {
+                    Action.VIEW_WILL_APPEAR -> {
+                        lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_START)
+
+                        if (isApplicationForeground) {
+                            lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
+                            Running(lifecycle = lifecycle)
+                        } else {
+                            Suspended(lifecycle = lifecycle)
+                        }
+                    }
+
+                    Action.VIEW_DID_DISAPPEAR -> {
+                        if (isApplicationForeground) {
+                            lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE)
+                        }
+
+                        lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
+
+                        Created(isApplicationForeground = isApplicationForeground, lifecycle = lifecycle)
```

Comment:
```suggestion
                        this
```

It doesn't sound right to emit Pause/Stop from Created state. If we in this state, it means we didn't receive `VIEW_WILL_APPEAR`, and so - didn't emit Start/Resume before.